### ISSUE TITLE
Jackson Processor for RangeSet

### ISSUE THREAD

**Issue by mittalnanu** [FasterXML/jackson-datatype-guava#90]
Are there any alternatives, I don't see any utility in the current module?

**Comment by cowtowncoder** [comment]
@mittalnanu um, what?

**Comment by mittalnanu** [comment]
Sorry, should have been more elaborate.
Let me try to explain using code
//Code 1
ObjectMapper mapper = new ObjectMapper();
 mapper.registerModule(new GuavaModule());
 System.out.println(mapper.writeValueAsString(Range.open(1,23)));

//Output 1
{"lowerEndpoint":1,"lowerBoundType":"OPEN","upperEndpoint":23,"upperBoundType":"OPEN"}

//Code 2 
RangeSet<Integer> rangeSet= TreeRangeSet.create();
rangeSet.add(Range.open(1,5));
rangeSet.add(Range.openClosed(7, 1000));
ObjectMapper mapper = new ObjectMapper();
mapper.registerModule(new GuavaModule());
System.out.println(mapper.writeValueAsString(rangeSet));

//Output 2
{"empty":false}

My question is RangeSet, doesn't seem to be supported out of the box. Is there something I am missing or need to write own Jackson Serializer / Deserializer ?

**Comment by cowtowncoder** [comment]
Module supports a subset of Guava datatypes, based on what users have needed, and contributed handlers for. Due to sheer size of Guava (very much monolithic collection of tons of things), coverage is not complete.

I do know that `Range` is supported, but I don't think `RangeSet` is.

**Comment by mittalnanu** [comment]
Thanks for the revert @cowtowncoder 
I found an alternative way to do it, is that how it should be done or we need to have explicit support for Rangeset?

Input 
RangeSet<Integer> rangeSet= TreeRangeSet.create();
rangeSet.add(Range.open(1,5));
rangeSet.add(Range.openClosed(7, 1000));
ObjectMapper mapper = new ObjectMapper();
mapper.registerModule(new GuavaModule());
System.out.println (mapper.writerFor(new TypeReference<Set>() {}).writeValueAsString(rangeSet.asRanges()))


Output
[{"lowerEndpoint":1,"lowerBoundType":"OPEN","upperEndpoint":5,"upperBoundType":"OPEN"},{"lowerEndpoint":7,"lowerBoundType":"OPEN","upperEndpoint":1000,"upperBoundType":"CLOSED"}]

**Comment by cowtowncoder** [comment]
@mittalnanu `RangeSet` is a type that does not seem to implement any of standard JDK collection types (`java.util.Collection` or `java.util.Set`), so the only way Jackson can handle it would be as regular POJO via getters and setters. And since it doesn't seem to define getters/setters, that's unlikely to work. This is why specific serializer and/or deserializer would be needed.

If `RangeSet` did actually implement `Set<Range>`, it would be handled automatically without any extra work.

**Comment by mittalnanu** [comment]
Yes so , rangeSet.asRanges() returns a Set<Range> and it works automatically. So I think that is a neat workaround. Do you see any scenario where a specific serializer is needed given I can get Set<Range> from RangeSet ?

**Comment by cowtowncoder** [comment]
@mittalnanu Right if and when you can get a regular `Set`, that works nice. Or you could even use mix-in annotations to attach `@JsonValue` to method `asRanges()` -- that would add automatic support for serialization.
